### DISCUSSION TITLE
Make this an official integration

### DISCUSSION THREAD

**Discussion by moskovskiy82** [ekutner/home-connect-hass#96] [General]
Just wondering why this hasn't been as an official integration instead of the basic "built in" one.

**Comment by ekutner** [comment]
Replacing the built-in integration with a different one that is not backward compatible is going to break the implementation of existing users. That is going to create a problem that I didn't want to deal so I didn't want to even raise it for consideration of the HA committers. Anyway, before it is even considered it should be demonstrated that this integration is better (I think so but that is not enough) and it is preferred by most users of the built-in integration.

**Comment by Nazze** [comment]
I love the integration and it is - at least in my case - much better than the built-in one.
HA gives the possibility to do config migrations, and there have been a lot of breaking changes in HA in the past. So, I really hope that a lot of people will be using it and it becomes part of the official package.

**Comment by kroimon** [comment]
I would also love to see PRs against the built-in integration instead of using a custom_component.
Breaking changes happen all the time, so that is not an issue as long as they improve the component.

Please make sure to break down your PRs in reviewable chunks. One single PR with a ton of changes at once will most likely not be accepted.

**Comment by ekutner** [comment]
@kroimon That's not going to happen. This is not an incremental update of the original integration but a new one built completely from scratch with a different approach. I have no idea what it will take to replace the built in one and, for the foreseeable future, don't expect to even try to find out.

**Comment by moskovskiy82** [comment]
Breaking changes? Which ones? The official integration can only inform on the status. So rewriting automations will be quite a breeze for the users

**Comment by ekutner** [comment]
All of them, this integration is completely different so anything that was built for the built-in one, UI wise or automations will not work with this one.

**Comment by bkbartk** [comment]
As far as I know, no new integrations with yaml configuration will be accepted by the HA team. So in order to make this one official this part needs to be rewritten.
I won't be so sure the breaking changes part would be an issue. I, and probably others have a lot of stability issues with the official integration. So when it comes to that. for me it was a choice of a non working integration or modify my scripts.
And after migrating, I like this one more, basically because of the statusses which are combined in a dropdown, instead of only a list with toggles, this one has more potential.

on the other hand, a HACS integration has the benefit that it can update outside the HA release cycle. This can be helpfull for critical issue as well as reacting on API changes.

Of course it would be up to @ekutner together with the HA developers. But is it was up to me, I would like to see this one as the official one.

**Comment by ChristophCaina** [comment]
it will!
I have used the official integration with my coffeemaker...

Just an example how it is working?
![grafik](https://user-images.githubusercontent.com/26391061/197135317-e68e545c-9033-41fd-95a5-dd3bf82e6e64.png)

This is everything the official integration does provide.
A switch - to turn the device on, some buttons for the program selection.
It does also provide the program progress.

And this is my new system with the HomeConnect Alt:
![grafik](https://user-images.githubusercontent.com/26391061/197135626-e0cd3eae-0f60-4969-af84-4c7964f5de72.png)

most parameters are now selects ... in case, the HomeConnect integration would just be replaced in the core - the whole dashboard  would become invalid and not functional anymore... at least, until the integration could "link" the existing buttons to its own triggers and what else.

I can see why this wish came up - and yes, I also agree - it would be nice to have the integration available in the core -  but I also do see the risk and the complains from others when their configs will fail

**Reply by moskovskiy82** [reply]
> it will! I have used the official integration with my coffeemaker...

I mean like writing a script for 3 or 5 buttons with setting input_select and putting them to lovelace... Are you sure that for those who will complain HA is the right choice? ) 

I mean with HA and this integration you could have less buttons

"Cappuccino ala Christoff"

Cappuccino
Extra Inensiv
Wassemenge 100ml (not sure what that means)
Temperatur 96
Start

"Cappuccino ala Christoff spouse"
Cappuccino
Weak
Wassemenge 200ml
Temperatur 80
Start

Think about that ;)

 It's as simple as 

    action:
      - service: input_select.select_option
        target:
          entity_id: input_select.coffe_bosche
        data:
          option: Capuccino
      - service: input_select.select_option
        target:
          entity_id: input_select.coffe_bosche
        data:
          option: Intensiv

**Reply by ChristophCaina** [reply]
With your examples, you don't even have to create buttons - adding a script with specific parameters and then trigger it via voice command ;)

btw: "Wassermenge" is the amount of water the machine will use.
If you have large cups, 220 to 240 ml would be fine - if you do have smaller cups, just use 190 ml ;)
